### PR TITLE
DOCSP-42339 Compound Indexes

### PR DESCRIPTION
**Pull request by lindseymoore**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-scala/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-42339>
Staging - <https://deploy-preview-74--docs-scala.netlify.app/indexes/compound-index/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/indexes/compound-index.txt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about any of the methods discussed in this guide, see the following API
+documentation:
+
+- `Indexes.compoundIndex() <{+api+}/org/mongodb/scala/model/Indexes$.html#compoundIndex(indexes:org.mongodb.scala.bson.conversions.Bson*):org.mongodb.scala.bson.conversions.Bson>`__
+- `MongoCollection.createIndex() <{+api+}/org/mongodb/scala/MongoCollection.html#createIndex(key:org.mongodb.scala.bson.conversions.Bson,options:org.mongodb.scala.model.IndexOptions):org.mongodb.scala.SingleObservable[String]>`__
+- `MongoCollection.find() <{+api+}/org/mongodb/scala/MongoCollection.html#find[C](filter:org.mongodb.scala.bson.conversions.Bson)(implicite:org.mongodb.scala.bson.DefaultHelper.DefaultsTo[C,TResult],implicitct:scala.reflect.ClassTag[C]):org.mongodb.scala.FindObservable[C]>`__
+- `MongoCollection.listIndexes() <{+api+}/org/mongodb/scala/MongoCollection.html#listIndexes[C]()(implicite:org.mongodb.scala.bson.DefaultHelper.DefaultsTo[C,org.mongodb.scala.Document],implicitct:scala.reflect.ClassTag[C]):org.mongodb.scala.ListIndexesObservable[C]>`__
```
Comment (mcmorisi) [inline source/indexes/compound-index.txt:118]: Nit: 

```suggestion
- `compoundIndex() <{+api+}/org/mongodb/scala/model/Indexes$.html#compoundIndex(indexes:org.mongodb.scala.bson.conversions.Bson*):org.mongodb.scala.bson.conversions.Bson>`__
- `createIndex() <{+api+}/org/mongodb/scala/MongoCollection.html#createIndex(key:org.mongodb.scala.bson.conversions.Bson,options:org.mongodb.scala.model.IndexOptions):org.mongodb.scala.SingleObservable[String]>`__
- `find() <{+api+}/org/mongodb/scala/MongoCollection.html#find[C](filter:org.mongodb.scala.bson.conversions.Bson)(implicite:org.mongodb.scala.bson.DefaultHelper.DefaultsTo[C,TResult],implicitct:scala.reflect.ClassTag[C]):org.mongodb.scala.FindObservable[C]>`__
- `listIndexes() <{+api+}/org/mongodb/scala/MongoCollection.html#listIndexes[C]()(implicite:org.mongodb.scala.bson.DefaultHelper.DefaultsTo[C,org.mongodb.scala.Document],implicitct:scala.reflect.ClassTag[C]):org.mongodb.scala.ListIndexesObservable[C]>`__
```

**Review by mcmorisi — CHANGES_REQUESTED** [review]
Good start! Would double check the code to make sure it's running.

File: source/indexes/compound-index.txt (hunk 1 above)
Comment (mcmorisi) [reply source/indexes/compound-index.txt:126]: I'd say my personal preference would be only to include the object class name if there were multiple methods with the same name in different object classes (e.g. if there were two methods named `MongoCollection.functionName()` and `MongoClient.functionName()`.). I don't think there's an explicit standard to do it one way or the other, so I'll leave it to your judgment!

Comment (lindseymoore) [reply source/indexes/compound-index.txt:126]: Would prefer to keep as is, but seems to be the standard in other docs sets like [pymongo](https://www.mongodb.com/docs/languages/python/pymongo-driver/current/databases-collections/#api-documentation) and [kotlin sync](https://www.mongodb.com/docs/languages/kotlin/kotlin-sync-driver/current/aggregation/#api-documentation) to not include. I'll remove for consistency.

**Review by mcmorisi — APPROVED** [review]
LGTM, nice job!

**Review by jyemin — APPROVED** [review]
Same issue as on previous PR with printResults, but otherwise LGTM

File: source/includes/indexes/compound-field.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// start-compound-index-imports
+import org.mongodb.scala._
+import org.mongodb.scala.model.Indexes
+import org.mongodb.scala.model.IndexOptions._
+import org.mongodb.scala.model.Projections._
+import org.mongodb.scala.model.Sorts._
+import org.mongodb.scala.model.Filters._
```
[the remark this answers is not in the document]
Comment (lindseymoore) [reply source/includes/indexes/compound-field.scala:7]: No longer using Helpers for consistency within this docs set and the other java docs.

**Pull request merged**